### ISSUE TITLE
Missing requirement of typing-extensions on python <= 3.10

### ISSUE THREAD

**Issue by rokm** [mdomke/schwifty#203]
On python versions that do not provide `typing.Self` (i.e., python <= 3.10), `schwifty` falls back to using `typing_extensions`. However, it does not specify `typing-extensions` as a requirement, and consequently user installing `schwifty` in a clean virtual environment will end up with defunct version. For example, on linux:

```
python3.10 -m venv venv
. venv/bin/activate
pip install schwifty
python -c "import schwifty" 
```

```
Traceback (most recent call last):
  File "[...]/venv/lib64/python3.10/site-packages/schwifty/common.py", line 9, in <module>
    from typing import Self
ImportError: cannot import name 'Self' from 'typing' (/usr/lib64/python3.10/typing.py)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "[...]/venv/lib64/python3.10/site-packages/schwifty/__init__.py", line 6, in <module>
    from schwifty.bban import BBAN
  File "[...]/venv/lib64/python3.10/site-packages/schwifty/bban.py", line 5, in <module>
    from schwifty import common
  File "[...]/lib64/python3.10/site-packages/schwifty/common.py", line 11, in <module>
    from typing_extensions import Self
ModuleNotFoundError: No module named 'typing_extensions'
```

**Comment by rokm** [comment]
Looks like https://github.com/mdomke/schwifty/commit/aa251c25c3439d5c2de476f444f5be7467ea866b accidentally removed ˙"typing_extensions>=4.0; python_version <= '3.11'"` requirement that was added in #201.

**Comment by mdomke** [comment]
I published a new release `2024.05.2` to address the issue.
